# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/Error.inl
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/Error.inl
--- a/gfx/angle/checkout/src/libANGLE/Error.inl	Sat Sep 07 13:36:46 2019 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,91 +0,0 @@
-//
-// Copyright (c) 2014 The ANGLE Project Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-// Error.inl: Inline definitions of egl::Error and gl::Error classes which encapsulate API errors
-// and optional error messages.
-
-#include "common/angleutils.h"
-
-#include <cstdarg>
-
-namespace egl
-{
-
-Error::Error(EGLint errorCode)
-    : mCode(errorCode),
-      mID(0)
-{
-}
-
-Error::Error(const Error &other)
-    : mCode(other.mCode),
-      mID(other.mID)
-{
-    if (other.mMessage)
-    {
-        createMessageString();
-        *mMessage = *(other.mMessage);
-    }
-}
-
-Error::Error(Error &&other)
-    : mCode(other.mCode),
-      mID(other.mID),
-      mMessage(std::move(other.mMessage))
-{
-}
-
-Error &Error::operator=(const Error &other)
-{
-    mCode = other.mCode;
-    mID = other.mID;
-
-    if (other.mMessage)
-    {
-        createMessageString();
-        *mMessage = *(other.mMessage);
-    }
-    else
-    {
-        mMessage.release();
-    }
-
-    return *this;
-}
-
-Error &Error::operator=(Error &&other)
-{
-    if (this != &other)
-    {
-        mCode = other.mCode;
-        mID = other.mID;
-        mMessage = std::move(other.mMessage);
-    }
-
-    return *this;
-}
-
-EGLint Error::getCode() const
-{
-    return mCode;
-}
-
-EGLint Error::getID() const
-{
-    return mID;
-}
-
-bool Error::isError() const
-{
-    return (mCode != EGL_SUCCESS);
-}
-
-// Static
-Error Error::NoError()
-{
-    return Error(EGL_SUCCESS);
-}
-
-}